core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -276,28 +276,26 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
   override def currentEventsByPersistenceId(
       persistenceId: String,
       fromSequenceNr: Long,
-      toSequenceNr: Long): Source[ClassicEventEnvelope, NotUsed] = {
-    val highestSeqNrFut =
-      if (toSequenceNr == Long.MaxValue) journalDao.readHighestSequenceNr(persistenceId, fromSequenceNr)
-      else Future.successful(toSequenceNr)
-
-    Source
-      .futureSource[SerializedJournalRow, NotUsed] {
-        highestSeqNrFut.map { highestSeqNr =>
-          internalEventsByPersistenceId(persistenceId, fromSequenceNr, highestSeqNr)
-        }
-      }
+      toSequenceNr: Long): Source[ClassicEventEnvelope, NotUsed] =
+    internalCurrentEventsByPersistenceId(persistenceId, fromSequenceNr, toSequenceNr)
       .map(deserializeRow)
-      .mapMaterializedValue(_ => NotUsed)
-  }
+
+  @ApiMayChange
+  def currentEventsByPersistenceIdTyped[Event](
```

Comment:
This is a good start, but I'm afraid we will need a new query interface in Akka, because we need it in Projections without dependency to r2dbc. See https://github.com/akka/akka-projection/pull/816/files#diff-86d4e7a7b1a3c630c066b698442916f3338ba622da7bea05f82719750f5906c0R238

Reply:
yeah, I have no better suggestion. The EventEnvelope is typded and even though that is in a different package name we must use unique names for the traits and methods because we must be able to mixin both old and new.